dot/peerset/peerstate.go

```diff
@@ -180,69 +181,99 @@ func (ps *PeersState) peers() []peer.ID {
 
 // sortedPeers returns the list of peers we are connected to of a specific set.
 func (ps *PeersState) sortedPeers(idx int) peer.IDSlice {
-	if len(ps.sets) < idx {
+	ps.RLock()
+	defer ps.RUnlock()
+
+	if len(ps.sets) == 0 || len(ps.sets) < idx {
 		logger.Debug("peer state doesn't have info for the provided index")
 		return nil
 	}
 
-	type kv struct {
-		peerID peer.ID
-		Node   *node
+	type connectedPeerReputation struct {
+		peerID     peer.ID
+		reputation Reputation
 	}
 
-	var ss []kv
-	for k, v := range ps.nodes {
-		state := v.state[idx]
+	connectedPeersReps := make([]connectedPeerReputation, 0)
+
+	for peerID, node := range ps.nodes {
+		state := node.state[idx]
+
 		if isPeerConnected(state) {
-			ss = append(ss, kv{k, v})
+			connectedPeersReps = append(connectedPeersReps, connectedPeerReputation{
+				peerID:     peerID,
+				reputation: node.reputation,
+			})
 		}
 	}
 
-	sort.Slice(ss, func(i, j int) bool {
-		return ss[i].Node.rep > ss[j].Node.rep
+	sort.Slice(connectedPeersReps, func(i, j int) bool {
+		return connectedPeersReps[i].reputation > connectedPeersReps[j].reputation
 	})
 
-	peerIDs := make(peer.IDSlice, len(ss))
-	for i, kv := range ss {
+	peerIDs := make(peer.IDSlice, len(connectedPeersReps))
+	for i, kv := range connectedPeersReps {
 		peerIDs[i] = kv.peerID
 	}
 
 	return peerIDs
 }
 
-func (ps *PeersState) addReputation(pid peer.ID, change ReputationChange) (
+func (ps *PeersState) updateReputationByTick(peerID peer.ID) (after Reputation, err error) {
+	ps.Lock()
+	defer ps.Unlock()
+
+	node, has := ps.nodes[peerID]
+	if !has {
+		return 0, ErrPeerDoesNotExist
+	}
+
+	after = reputationTick(node.reputation)
```

Comment:
nit maybe rename `after` to `newReputation`? I thought it was some kind of time at a quick glance.